### PR TITLE
pkg/version.go: mod CheckClusterVersion: fix StatusCode check issue

### PR DESCRIPTION
**Pull request by JinLingChristopher**
… make code easy to read

### What problem does this PR solve? 

* fix wrong http status code in pkg/version.go `CheckClusterVersion` function

### What is changed and how it works?

* fix status logic
* add space to make code much more easy to read

### Check List 

Tests 

 - No code

Code changes

- alter logic opeartion `&&` -> `||`

Side effects

- No side effects

Related changes

- No other related changes

### Release note

No release note

### CODE REVIEW

File: pkg/version/check.go [hunk 1]
```diff
@@ -70,35 +71,42 @@ func CheckClusterVersion(
...
 	if err != nil {
 		return cerror.WrapError(cerror.ErrCheckClusterVersionFromPD, err)
 	}
+
 	resp, err := httpCli.Do(req)
 	if err != nil {
 		return cerror.WrapError(cerror.ErrCheckClusterVersionFromPD, err)
 	}
-	if resp.StatusCode < 200 && resp.StatusCode >= 300 {
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```
Comment (amyangfei) [inline pkg/version/check.go:85]: Could you please add a unit test for this?

**Comment by amyangfei** [comment]
please fix `make check`, rest lgtm

**Review by amyangfei — APPROVED** [review]

**Comment by JinLingChristopher** [comment]
This is so fxxking weird.

**Comment by JinLingChristopher** [comment]
this ci is so weird

**Review by leoppro — APPROVED** [review]

**Comment by amyangfei** [comment]
The CI engineers are investigating why the `leak-test` is not running, we can wait and see. 😂 @JinLingChristopher

**Comment by amyangfei** [comment]
Find another bug from the leak test, https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/cdc_ghpr_leak_test/detail/cdc_ghpr_leak_test/2912/pipeline

The `resp.Body` is not closed in some scenarios, such as status code is not 2xx, decode response error, etc.

Would you please also take a look @JinLingChristopher

**Comment by JinLingChristopher** [comment]
defer `resp.Body.Close()`, no need to check the error in defer.

**Pull request merged**